tools/extend_sr.php

```diff
@@ -19,7 +20,8 @@ class ActionException extends Exception
 
 $title = _("Extend the Smooth Reading deadline");
 $new_state = PROJ_POST_FIRST_CHECKED_OUT;
-$back_url = "$code_url/project.php?id=$projectid&amp;expected_state=$new_state#smooth_start";
+$suffix = "&amp;expected_state=$new_state#smooth_start";
```

Comment:
if we took the query params, we could avoid this `& amp ;` stuff (spaces for github)

Reply:
Thanks @chrismiceli 
So, if I understand correctly, for this case, I would end up calling it something like this:
```
$query_param_array = array("expected_state=$new_state");
return_to_project_page_link($projectid, $query_param_array, "#smooth_start");
```
Is that what you're suggesting? Presumably also with `$query_param_array` and `$fragment` being optional arguments to `return_to_project_page_link`, since neither of them is needed for many calls?

Final question (for now): Do you also recommend allowing the function to be called with a single string in place of the array argument when only one query param is needed (most cases), i.e. 
```
return_to_project_page_link($projectid, "expected_state=$new_state", "#smooth_start");
```
I believe that's permitted in PHP if I check it inside the function, isn't it? Or do you think that is an unnecessary complication and prone to errors?

Reply:
I think the query-params-as-array approach is a more robust one. Probably better for the second arg to either be an array or Null and just have the caller wrap a singe arg in `[]`s, eg:
```php
return_to_project_page_link($projectid, ["expected_state=$new_state"], "#smooth_start");

```

Reply:
Yes, I like what cpeel has. And yes, if we can, support defaulting the 2nd and 3rd parameters since they aren't always needed. We could also avoid the `#` on the fragment argument since it will always be present.